scipy/spatial/tests/test_qhull.py

```diff
@@ -902,6 +903,105 @@ def simplified(x):
 
             yield check, name
 
+class Test_HalfspaceIntersection(object):
+    def assert_unordered_allclose(self, arr1, arr2, rtol=1e-7):
+        """Check that every line in arr1 is only once in arr2"""
+        assert_equal(arr1.shape, arr2.shape)
+
+        truths = np.zeros((arr1.shape[0],), dtype=bool)
+        for l1 in arr1:
+            indexes = np.where((abs(arr2 - l1) < rtol).all(axis=1))[0]
+            assert(indexes.shape == (1,))
+            truths[indexes[0]] = True
+        assert(truths.all())
+
+    def test_cube_halfspace_intersection(self):
+        halfspaces = np.array([[-1.0, 0.0, 0.0],
+                               [0.0, -1.0, 0.0],
+                               [1.0, 0.0, -1.0],
+                               [0.0, 1.0, -1.0]])
+        feasible_point = np.array([0.5, 0.5])
+
+        points = np.array([[0.0, 1.0], [1.0, 1.0], [0.0, 0.0], [1.0, 0.0]])
+
+        hull = qhull.HalfspaceIntersection(halfspaces, feasible_point)
+
+        assert_allclose(points, hull.intersections)
+
+    def test_self_dual_polytope_intersection(self):
+        fname = os.path.join(os.path.dirname(__file__), 'data',
+                             'selfdual-4d-polytope.txt')
+        ineqs = np.genfromtxt(fname)
+        halfspaces = -np.hstack((ineqs[:, 1:], ineqs[:, :1]))
+
+        hs = qhull.HalfspaceIntersection(halfspaces, np.array([0., 0., 0., 0.]))
+
+        assert_equal(hs.intersections.shape, (24, 4))
```

Comment:
You are right, but I was mostly interested in the intersection points, I'll see what I can do about this.